tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,182 @@
+from common import ProviderOptions, data_bytes
+from fixtures import managed_process # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL, GnuTLS
+from utils import invalid_test_parameters, get_parameter_name, to_bytes
+
+SEND_DATA_SIZE = 2 ** 14
+
+K_BYTES = 1024 
+SEND_BUFFER_SIZE_MIN = 1031
+SEND_BUFFER_SIZE_MIN_RECOMMENDED = 2 * K_BYTES
+SEND_BUFFER_SIZE_MULTI_RECORD = 17 * K_BYTES
+SEND_BUFFER_SIZE_PREFER_THROUGHPUT = 35 * K_BYTES
+SEND_BUFFER_SIZE_HUGE = 512 * K_BYTES
+
+SEND_BUFFER_SIZES = [
+    SEND_BUFFER_SIZE_MIN,
+    SEND_BUFFER_SIZE_MIN_RECOMMENDED,
+    SEND_BUFFER_SIZE_MULTI_RECORD,
+    SEND_BUFFER_SIZE_PREFER_THROUGHPUT,
+    SEND_BUFFER_SIZE_HUGE
+]
+
+FRAGMENT_PREFERENCE = [
+    None,
+    "--prefer-low-latency",
+    "--prefer-throughput"
+]
+
+TEST_CERTS = [
+    Certificates.RSA_2048_SHA256,
+    Certificates.RSA_PSS_2048_SHA256,
+    Certificates.ECDSA_256
+]
+
+def is_subsequence(s, t):
+    """predicate: is s a subsequence of t?
+
+    Parameters
+    ----------
+    s: bytes
+        subsequence to look for
+    t: str, bytes
+        sequence to look in
+
+    Returns
+    -------
+    bool
+        True if s is a subsequence, False if not 
+    """
+    s = str(s)[2:-1] # Remove the b' and ' from the converted bytes
+    t = str(t)
+    s_len, t_len = len(s), len(t)
+    s_index, t_index = 0, 0
+    while (s_index < s_len and t_index < t_len):
+        if (s[s_index] == t[t_index]):
+            s_index += 1
+        t_index += 1
+    return s_index == s_len
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [GnuTLS, OpenSSL, S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("buffer_size", SEND_BUFFER_SIZES, ids=get_parameter_name)
+@pytest.mark.parametrize("fragment_preference", FRAGMENT_PREFERENCE, ids=get_parameter_name)
+def test_s2n_buffered_send_server(managed_process, cipher, other_provider, provider, protocol, certificate, buffer_size, fragment_preference):
+    # Communication Timeline
+    # Client [S2N|OpenSSL|GnuTLS]  | Server [S2N]
+    # Handshake                    | Handshake
+    #  Handshake finish indicated
+    #  by the client send marker
+    # Send Server Send Marker      | Receive the Server Send Marker "CLIENTSENT"
+    #                              | Send Data Bytes to Client (stresses the send buffer - center of test)
+    #                              | Send Client Close Marker "SERVERSENT"
+    # Close                        | Close
+    
+    # for the corresponding provider, these strings appear when on stdout when the handshake is finished
+    starting_client_send_markers = {
```

Comment:
Could the send markers returned by `provider.get_send_marker` be used here?

Reply:
Maybe, I am checking now.

Reply:
Yes, indeed that works.